### PR TITLE
Add snmp communities to eos_cli_config_Gen

### LINKED ISSUE #201: Add snmp-server community to snmp-settings template in eos_cli_config_gen
**Issue Type**
- Template enhancement

**Is your feature request related to a problem? Please describe.**
The community option is missing from snmp-settings

**Describe the solution you'd like**
Update to the following:
```
snmp_server:
  contact: < contact_name >
  location: < location >
  local_interfaces:
    - name: < interface_name_1 >
      vrf: < vrf_name >
    - name: < interface_name_2 >
  views:
    - name: < view_name >
      MIB_family_name: < MIB_family_name >
      included: < true | false >
    - name: < view_name >
      MIB_family_name: < MIB_family_name >
      included: < true | false >
  communities:
    - < community_name_1 >:
       access: < ro | rw >
       access_list:
          type: < ipv4 | ipv6 >
          name: 
       view: <view_name >
     - < community_name_2 >:
       access: < ro | rw >
       access_list:
          type: < ipv4 | ipv6 >
          name: 
       view: <view_name >
  groups:
    - name: < group_name >
      version: < v1 | v2c | v3 >
      authentication: < auth | noauth | priv >
      read: < read_view >
      write: < write_view >
      notify: < notify_view >
    - name: < group_name >
      version: < v1 | v2c | v3 >
      authentication: < auth | noauth | priv >
      read: < read_view >
  users:
    - name: < username >
      group: < group_name >
      version: < v1 | v2c | v3 >
      auth: < hash_algorithm >
      auth_passphrase: < encrypted_auth_passphrase >
      priv: < encryption_algorithm >
      priv_passphrase: < encrypted_priv_passphrase >
    - name: < username >
      group: < group_name >
      version: < v1 | v2c | v3 >
  hosts:
    - host: < host IP address or name >
      vrf: < vrf_name >
      users:
        - username: < username >
          authentication_level: < auth | noauth | priv >
          version: < 1 | 2c | 3 >
    - host: < host IP address or name >
      vrf: < vrf_name >
      users:
        - username: < username >
          authentication_level: < auth | noauth | priv >
          version: < 1 | 2c | 3 >
  traps:
    enable: < true | false >
  vrfs:
    - name: < vrf_name >
      enable: < true | false >
    - name: < vrf_name >
      enable: < true | false >
```

**Describe alternatives you've considered**
N/A

**Additional context**
Open to adjusting the model as this data model is already quite long / complex compared to others.

### PR DESCRIPTION
**Pull request by melkypie**
## Change Summary

Adds snmp-server communities to eos_cli_config_gen

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [X] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #201 
Supersedes #393

## Component(s) name

`arista.avd.eos_cli_config_Gen`

## Proposed changes

Changes the snmp_server data model to include communities:
```yaml
snmp_server:
  contact: < contact_name >
  location: < location >
  communities:
    - name: < community_name_1 >
      access: < ro | rw >
      access_list_ipv4:
        name: < acl_ipv4_name >
      access_list_ipv6:
        name: < acl_ipv6_name >
      view: < view_name >
    - name: < community_name_2 >
      access: < ro | rw >
      access_list_ipv4:
        name: < acl_ipv4_name >
      access_list_ipv6:
        name: < acl_ipv6_name >
      view: < view_name >
```

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [X] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by carlbuchmann** [comment]
@melkypie Thanks for submitting this PR!

Please update the data model to the following:

```yaml
snmp_server:
  communities:
    < community_name_1 >:
      access: < ro | rw >
      access_list_ipv4:
        name: < acl_ipv4_name >
      access_list_ipv6:
        name: < acl_ipv6_name >
      view: <view_name >
    < community_name_2 >:
      access: < ro | rw >
      access_list_ipv4:
        name: < acl_ipv4_name >
      access_list_ipv6:
        name: < acl_ipv6_name >
      view: <view_name >
```

**Comment by melkypie** [comment]
Seems like the CI failed on a completely unrelated thing. This  https://www.avd.sh/en/releases-v2.0.x/ is throwing a 404 error.

**Comment by titom73** [comment]
Hello @melkypie 

This error can be safely ignored and will be fixed in an additional PR.
It is related to a change in branch name we did last week and not reported back to documentation.

**Comment by titom73** [comment]
For information, the link issue has been fixed with the following PR https://github.com/aristanetworks/ansible-avd/pull/846

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2 [hunk 1]
```diff
@@ -22,6 +22,31 @@
 {%         endfor %}
 {%     endif %}
 
+{% if snmp_server.communities is arista.avd.defined %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2]: @melkypie - we are still in process of refactoring our documentation templates to follow a certain style. 
We plan on publishing a guide soon to give better guidance for external contributors. 

Can you update the jinja2 template to follow this style from the following examples:

- https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2
- https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/documentation/fabric-documentation.j2

**Review by carlbuchmann — COMMENTED** [review]
Please see the comment regarding the jinja2 template style for documentation. The rest is perfect!

**Review by carlbuchmann — APPROVED** [review]
LGTM - and thank you for this contribution!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/snmp-settings.j2 [hunk 2]
```diff
@@ -7,6 +7,26 @@ snmp-server contact {{ snmp_server.contact }}
...
+{%             endif %}
+{%             if snmp_server.communities[community].access_list_ipv6 is arista.avd.defined %}
+{%                 set communities_cli = communities_cli ~ " ipv6 " ~ snmp_server.communities[community].access_list_ipv6.name %}
+{%             endif %}
+{%             if snmp_server.communities[community].access_list_ipv4 is arista.avd.defined %}
+{%                 set communities_cli = communities_cli ~ " " ~ snmp_server.communities[community].access_list_ipv4.name %}
+{%             endif %}
+{{ communities_cli }}
+{%         endfor %}
+{%     endif %}
```
Comment (xaviramon) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/snmp-settings.j2]: This should be rendered between "view" and "group" commands.
Tested on EOS 4.23.6M:
```eos
snmp-server view VW-WRITE iso included
snmp-server community private rw
snmp-server community public ro
snmp-server community snmp-test-comm ro onur
snmp-server group GRP-READ-ONLY v3 priv read v3read
```

**Comment by carlbuchmann** [comment]
@melkypie - Thank you for this contribution!

**Pull request merged**